source/compiler-core/slang-json-parser.h

```diff
@@ -0,0 +1,188 @@
+// slang-json-parser.h
+#ifndef SLANG_JSON_PARSER_H
+#define SLANG_JSON_PARSER_H
+
+#include "slang-json-lexer.h"
+
+
+namespace Slang {
+
+class JSONListener
```

Comment:
I'm on the fence about the "listener" sort of approach as opposed to just committing to always building some kind of AST.

This design is clearly a big win for building up the infrastructure in pieces, since having the `JSONWriter` provides a way to test that the parser is working even before implementing the AST.